### PR TITLE
RavenDB-12811 dashboard updates for non cluster admin operators

### PR DESCRIPTION
**Pull request by ml054**

### CODE REVIEW

File: src/Raven.Server/Dashboard/DatabasesInfo.cs [hunk 1]
```diff
@@ -21,6 +22,18 @@ public override DynamicJsonValue ToJson()
             json[nameof(Items)] = new DynamicJsonArray(Items.Select(x => x.ToJson()));
             return json;
         }
+
+        public override DynamicJsonValue ToJsonWithFilter(Func<string, bool> filter)
+        {
+            var items = new DynamicJsonArray(Items.Where(x => filter(x.Database)).Select(x => x.ToJson()));
```
Comment (ppekrol) [inline src/Raven.Server/Dashboard/DatabasesInfo.cs]: let's avoid LINQ, some of these notifications are created often when Studio is connected

Comment (ml054) [reply src/Raven.Server/Dashboard/DatabasesInfo.cs]: done

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Server/Dashboard/DatabasesInfo.cs [hunk 2]
```diff
@@ -21,6 +22,25 @@ public override DynamicJsonValue ToJson()
             json[nameof(Items)] = new DynamicJsonArray(Items.Select(x => x.ToJson()));
             return json;
         }
+
+        public override DynamicJsonValue ToJsonWithFilter(Func<string, bool> filter)
```
Comment (ayende) [inline src/Raven.Server/Dashboard/DatabasesInfo.cs:26]: Why not return an empty array?

File: src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs [hunk 3]
```diff
@@ -56,18 +55,6 @@ public async Task WriteNotifications(Func<string, bool> shouldWriteByDb)
                             continue;
                         }
 
-                        if (shouldWriteByDb != null &&
-                            shouldWriteByDb((string)tuple.Item2["Database"]) == false)
-                        {
-                            if (sp.ElapsedMilliseconds > 5000)
-                            {
-                                sp.Restart();
-                                await SendHeartbeat();
```
Comment (ayende) [inline src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs:65]: We actually probably need this here. If we have a lot of events / messages, but none that we can send, we need to do the heartbeat here.

File: src/Raven.Server/Dashboard/DatabasesInfo.cs (hunk 2 above)
Comment (ml054) [reply src/Raven.Server/Dashboard/DatabasesInfo.cs:26]: if we return null, then we don't append anything to the queue

File: src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs (hunk 3 above)
Comment (ml054) [reply src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs:65]: see note above - when we return null such message is not queued, so we timeout on queue.get and heartbird is sent

File: src/Raven.Server/Dashboard/DatabasesInfo.cs (hunk 2 above)
Comment (ml054) [reply src/Raven.Server/Dashboard/DatabasesInfo.cs:26]: moreover we need to filter inner structures and we don't want to send database names users can't access.

File: src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs (hunk 3 above)
Comment (ml054) [reply src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs:65]: here: https://github.com/ravendb/ravendb/pull/8837/files/a22b9775326dfcd390fe541d5d948d731f9990ac#diff-88c47ac394f9600c53afb59060c78863L52

Comment (ayende) [reply src/Raven.Server/NotificationCenter/NotificationCenterWebsocketWriter.cs:65]: Okay, looks good, then

**Pull request merged**